libs/template/config.go

```diff
@@ -0,0 +1,198 @@
+package template
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/jsonschema"
+)
+
+type config struct {
+	ctx    context.Context
+	values map[string]any
+	schema *jsonschema.Schema
+}
+
+func newConfig(ctx context.Context, schemaPath string) (*config, error) {
+	// Read config schema
+	schemaBytes, err := os.ReadFile(schemaPath)
+	if err != nil {
+		return nil, err
+	}
+	schema := &jsonschema.Schema{}
+	err = json.Unmarshal(schemaBytes, schema)
+	if err != nil {
+		return nil, err
+	}
+
+	// Return config
+	return &config{
+		ctx:    ctx,
+		schema: schema,
+		values: make(map[string]any, 0),
+	}, nil
+}
+
+// Reads json file at path and assigns values from the file
+func (c *config) assignValuesFromFile(path string) error {
+	// Read the config file
+	configFromFile := make(map[string]any, 0)
+	b, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(b, &configFromFile)
+	if err != nil {
+		return err
+	}
+
+	// Cast any integer properties, from float to integer. Required because
+	// the json unmarshaller treats all json numbers as floating point
+	for name, floatVal := range configFromFile {
+		property, ok := c.schema.Properties[name]
+		if !ok {
+			return fmt.Errorf("%s is not defined as an input parameter for the template", name)
+		}
+		if property.Type != jsonschema.IntegerType {
+			continue
+		}
+		v, err := toInteger(floatVal)
+		if err != nil {
+			return fmt.Errorf("failed to cast value %v of property %s from file %s to an integer: %w", floatVal, name, path, err)
+		}
+		configFromFile[name] = v
+	}
```

Comment:
Maybe. If a custom json unmarshaller we implement treats integer valued numbers as an `int` when type is ambigious, then we would not need this. Ofcourse doing this would have other side effects.